### DISCUSSION TITLE
No possibility to send the Open Push Notification event manually

### DISCUSSION THREAD

**Discussion by c-karolpawlik** [Iterable/iterable-android-sdk#433] [Q&A] [answered]
Hi,

I'm using the Iterable SDK. And we have configured push notifications  as OS Notification [(the Iterable doc)](https://support.iterable.com/hc/en-us/articles/115000331943-Setting-up-Android-Push-Notifications-#new-push-integration)
And I need to send the event about clicking on the Iterable push notification in system tray.
I handle it according to [firebase doc](https://firebase.google.com/docs/cloud-messaging/android/receive#backgrounded).

I noticed that you have in your SDK methods `trackPushOpen` inside class `IterableApi`. These methods are `protected`. Can change visibility to public and publish it for users?  
If yes, I'd like also have possibility to  prepare parameters for that method..  Do you have some methods which extracts parameters from the Intent for `trackPushOpen` method purposes?

My goal is  to get that event on WEB:
<img width="189" alt="image" src="https://user-images.githubusercontent.com/91550842/157060818-085aabe2-3c7a-4648-bfb9-ce519c4a5dcf.png">

I tried to send` IterableConstants.ACTION_NOTIF_OPENED` event with IterableApi.getInstance().track(IterableConstants.ACTION_NOTIF_OPENED,.....). But the result was:
<img width="323" alt="image" src="https://user-images.githubusercontent.com/91550842/157063447-e308a0ac-8b2c-4877-9862-32a788f3163c.png">

I'll be appreciate for your answer

**Comment by Ayyanchira** [comment, marked as answer]
Thanks for the suggestion 😇
Version [3.4.6](https://github.com/Iterable/iterable-android-sdk/releases/tag/3.4.6) has made `trackPushOpen` public. Hope it helps.

**Reply by c-karolpawlik** [reply]
Thank you for completing it.
